fragment.go

```diff
@@ -2669,6 +2669,8 @@ func filterWithRows(rows []uint64) rowFilter {
 // this container have been processed. The rows accumulated up to this point
 // (including this row if all filters passed) will be returned.
 func (f *fragment) rows(start uint64, filters ...rowFilter) []uint64 {
+	f.mu.Lock()
```

Comment:
RLock?

Reply:
... i was actually gonna file an issue for "we should use an RWMutex for this so we can RLock here", because I apparently never checked.